local-storage.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable consistent-return */
+function storageAvailable(type) {
+  let storage;
+  try {
+    storage = window[type];
+    const x = '__storage_test__';
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (e) {
+    return (
+      e instanceof DOMException
+      // everything except Firefox
+      && (e.code === 22
+        // Firefox
+        || e.code === 1014
+        // test name field too, because code might not be present
+        // everything except Firefox
+        || e.name === 'QuotaExceededError'
+        // Firefox
+        || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+      // acknowledge QuotaExceededError only if there's something already stored
+      && storage
+      && storage.length !== 0
+    );
+  }
+}
+
+function setFormValues() {
+  const currentName = localStorage.getItem('name');
+  const currentEmail = localStorage.getItem('email');
+  const currentMessage = localStorage.getItem('message');
+
+  document.getElementById('user-name').value = currentName;
+  document.getElementById('user-email').value = currentEmail;
+  document.getElementById('user-message').value = currentMessage;
+}
+
+function populateStorage() {
+  localStorage.setItem('name', document.getElementById('user-name').value);
+  localStorage.setItem('email', document.getElementById('user-email').value);
+  localStorage.setItem('message', document.getElementById('user-message').value);
+  setFormValues();
+}
+
+if (storageAvailable('localStorage')) {
+  if (localStorage.length < 1) {
+    populateStorage();
+  } else {
+    setFormValues();
+  }
+} else {
+  return false;
+}
+
+document.getElementById('user-name').onchange = populateStorage;
+document.getElementById('user-email').onchange = populateStorage;
+document.getElementById('user-message').onchange = populateStorage;
```

Comment:
- [ ] Currently, saving and retrieving form data from local storage is not working, looking at the video below, you can see there is an error on the console which prevents sit from working. The error points to line `53` of this file, I'll suggest you take a look at this.  :+1: :100: 

- Video reference:

https://github.com/Zilola-Nazarova/zilola-portfolio/assets/107632780/b2419e49-abf0-4a94-8adf-8b1b264162a3

- [ ] Also I'll suggest using the event of `keypress` instead of the `onchange` event, The onchage event requires the user to leave the input field after typing before it could save the data, but onpress event will save the user input at each key press thereby having the users latest data even if the user refreshes the page without clicking out of the input field. :+1: :+1:  You can read more on [this event here](https://www.w3schools.com/jsref/event_onkeypress.asp). :smile: :small_airplane: